### PR TITLE
fix k8s version issue

### PR DESCRIPTION
**Pull request by nicogbg**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Use k8s version file when building k8s RPMs 

###### Change Log  

relevant k8s spec and add version files

###### Does this affect the toolchain?  

NO

### CODE REVIEW

File: SPECS/kubernetes/version-file-1.18.14.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+KUBE_GIT_COMMIT='1adde27c8a2253fde5290a6874e9cdb6cb318fc0'
```
Comment (PawelWMS) [inline SPECS/kubernetes/version-file-1.18.14.sh:1]: What do you think about keeping one template script for which we'd replace the commit and version info inside the spec before it's passed to the build? That way we wouldn't need to keep track of these files (add new ones, remove the old) and update everything in one place - the spec file?

It should simplify the updates automation a lot as well.

Comment (PawelWMS) [reply SPECS/kubernetes/version-file-1.18.14.sh:1]: Alternatively, we could use something like this in the spec and skip checking in the script completely:

```
cat > version-file.sh << EOF
KUBE_GIT_COMMIT='%{source_commit_hash}'
KUBE_GIT_TREE_STATE='archive'
KUBE_GIT_VERSION='v%{version}-%{hotfix_number}'
KUBE_GIT_MAJOR='%{version_major}'
KUBE_GIT_MINOR='%{version_minor}+'
EOF
```

The placeholders would be regular RPM macros defined in the spec.

Comment (nicogbg) [reply SPECS/kubernetes/version-file-1.18.14.sh:1]: by using a template we kind of infer the versioning rules from the tarball name. The real source of trust is version.sh (in k8s code) not what we think the version should be. So I prefer have the automation pipeline creating version file per k8s version we build.
The automation pipeline will have to use version.sh to create those files.

**Review by PawelWMS — APPROVED** [review]

File: SPECS/kubernetes/version-file-1.18.14.sh (hunk 1 above)
Comment (PawelWMS) [reply SPECS/kubernetes/version-file-1.18.14.sh:1]: Discussed offline, makes sense, closing comment.

**Pull request merged**